ConfigSpace/hyperparameters.pyx

```diff
@@ -877,12 +888,283 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
+                    % (self.upper, self.lower, self.q)
+                )
+
+    def __repr__(self) -> str:
+        repr_str = io.StringIO()
+        repr_str.write("%s, Type: BetaFloat, Alpha: %s Beta: %s, Range: [%s, %s], Default: %s" % (self.name, repr(self.alpha), repr(self.beta), repr(self.lower), repr(self.upper), repr(self.default_value)))
+
+        if self.log:
+            repr_str.write(", on log-scale")
+        if self.q is not None:
+            repr_str.write(", Q: %s" % str(self.q))
+        repr_str.seek(0)
+        return repr_str.getvalue()
+
+    def __eq__(self, other: Any) -> bool:
+        """
+        This method implements a comparison between self and another
+        object.
+
+        Additionally, it defines the __ne__() as stated in the
+        documentation from python:
+            By default, object implements __eq__() by using is, returning NotImplemented
+            in the case of a false comparison: True if x is y else NotImplemented.
+            For __ne__(), by default it delegates to __eq__() and inverts the result
+            unless it is NotImplemented.
+
+        """
+        if not isinstance(other, self.__class__):
+            return False
+
+        return (
+            self.name == other.name and
+            self.default_value == other.default_value and
+            self.alpha == other.alpha and
+            self.beta == other.beta and
+            self.log == other.log and
+            self.q == other.q and
+            self.lower == other.lower and
+            self.upper == other.upper
+        )
+
+    def __copy__(self):
+        return BetaFloatHyperparameter(
+            name=self.name,
+            default_value=self.default_value,
+            alpha=self.alpha,
+            beta=self.beta,
+            log=self.log,
+            q=self.q,
+            lower=self.lower,
+            upper=self.upper,
+            meta=self.meta
+        )
+
+    def __hash__(self):
+        return hash((self.name, self.alpha, self.beta, self.lower, self.upper, self.log, self.q))
+
+    def to_uniform(self) -> 'UniformFloatHyperparameter':
+        lb = self.lower
+        ub = self.upper
+
+        return UniformFloatHyperparameter(self.name,
+                                          lb,
+                                          ub,
+                                          default_value=self.default_value,
+                                          q=self.q, log=self.log)
+
+    def check_default(self, default_value: Union[int, float]) -> Union[int, float]:
+        # return mode as default
+        lb = self.lower
+        ub = self.upper
+        
+        if default_value is None:
+            if (self.alpha > 1) and (self.beta > 1):
+                normalized_mode = (self.alpha - 1) / (self.alpha + self.beta - 2)
+                return self._transform_scalar((self._upper - self._lower) * normalized_mode + self._lower)
+            else: 
+                return (ub - lb) / 2
```

Comment:
This happens if `alpha == 1` and `beta == 1`, right? Is this a special case of the `if` above or a special case of the beta distribution??

Reply:
"This happens if alpha == 1 and beta == 1, right?"

Correct. When you have these parameters, it is actually just a uniform distribution.

Reply:
Actually, it also happens if `alpha == 1 or beta == 1`. Would that be an issue?

Reply:
And could you maybe add a comment explaining this to the code?

Reply:
Yeah, sure!